### DISCUSSION TITLE
How can I change the progress of a playing sound?

### DISCUSSION THREAD

**Discussion by Ikutsu** [korlibs/korge#2215] [Q&A]
I want to change to progress of a SoundChannel and I have tried change the current and position value but it does not work.

**Comment by soywiz** [comment]
It should be done with `.position`. Can you provide detailes where it  is not working?

* Platform (JVM, JS, iOS, Android)
* Which Operating System / Browser (Windows, Linux, Mac, iOS 14, Android 11, Chrome, Firefox, etc.)
* Audio that is not working
* Snippet code reproducing the issue
* KorGE/KorAU version

**Reply by ArtyomZhukov** [reply]
Same problem happened to me.

I use "com.soywiz.korge:korge-core:5.4.0" for Multiplatform app (Android and Desktop)

It looks like `.position` is used to position the sound in 3d space of game scene.

![image](https://github.com/korlibs/korge/assets/20988586/d43d42d8-bf96-481c-aabf-785e45683373)

**Reply by Ikutsu** [reply]
I am using korge-core 5.4.0 in compose multiplatform for jvm and testing on windows.

When I use **startTime**
![图片](https://github.com/korlibs/korge/assets/73303056/68b6f50a-2950-4017-b7dd-acb743ebb06c)
I get
![图片](https://github.com/korlibs/korge/assets/73303056/b2c6673d-a83a-4764-86d7-f282aa0f2599)

When I use **current**
![图片](https://github.com/korlibs/korge/assets/73303056/67b4eff9-2333-4336-8c90-22e180e411be)
I get (after I call the function I reset to 0 second)
![图片](https://github.com/korlibs/korge/assets/73303056/d39dffc8-83f0-4a8a-bb9a-b9351b4b59ec)

When I use **position** (I don't know if this is the right way to use this)
![图片](https://github.com/korlibs/korge/assets/73303056/e05e3fe9-8e6a-489a-b94b-e7924789f769)
![图片](https://github.com/korlibs/korge/assets/73303056/f6b02c41-9a65-4df3-8cd1-5e338c3354fd)
Nothing happens 
![图片](https://github.com/korlibs/korge/assets/73303056/f050146b-b228-4997-9dc2-2a10f3518821)
![图片](https://github.com/korlibs/korge/assets/73303056/71ecb754-4623-4bf3-9854-a78a504bd07d)

**Reply by Ikutsu** [reply]
I think I found what the problem is. I used the debugger to track what the program does when I set `.current`. I see that it calls `createDecoderStream()` in `MP3Decoder`. It will later set the `currentPositionInSamples` and check if `mp3SeekingTable `is null or not. In the picture below, the `mp3SeekingTable` is `null`, so the statement went into the else branch, which will reset the `position` and `_currentPositionInSamples` and then reset the sound.
![image](https://github.com/korlibs/korge/assets/73303056/a4d304c1-374a-4afa-ad05-b4d8c3aadbcb)
![image](https://github.com/korlibs/korge/assets/73303056/de03c4d2-4e35-4120-a085-d2a103103226)

**Reply by Ikutsu** [reply]
I added true to exactTimings in AudioDecodingProps and it generates a mp3SeekingTable. But the song is still starting at the beginning, only the current value is starting from the time I set. 
![image](https://github.com/korlibs/korge/assets/73303056/03e8c6dd-6a21-4b1c-a26e-d0a1814a682e)

**Reply by Ikutsu** [reply]
I assume that the pcm did not get repositioned or regenerate(decode) ?

**Reply by Ikutsu** [reply]
I have tested with wav file, so I look like it is a mp3 decode problem.

**Reply by Ikutsu** [reply]
And it look like ogg files are not supported?
![image](https://github.com/korlibs/korge/assets/73303056/98fa5947-c196-4819-81ab-cac480c4480c)
![image](https://github.com/korlibs/korge/assets/73303056/919fd189-fc38-404a-b140-528112b805d1)

**Reply by soywiz** [reply]
Thanks for the details. Ogg Vorbis is not supported except in Kotlin/Native. Only MP3 and WAV are bundled. ([opus](https://store.korge.org/module/korau-opus/), [flac](https://store.korge.org/module/korau-flac/) and [other formats](https://store.korge.org/module/korau-mod/) are supported via store).
Can you provide the mp3 files and the full snippet in textual format playing and changing the position of it? So I can investigate? Also it would be better to be an issue instead of a discussion, so I can add it to the project roadmap board.
